### PR TITLE
Fixes #3921 - Make choice of discovery fact name a setting

### PR DESCRIPTION
**Pull request by GregSutcliffe**
@lzap thoughts? @mmol care to test drive? :)

### CODE REVIEW

**Comment by mmoll** [comment]
works for me :)

File: app/models/setting/discovered.rb [hunk 1]
```diff
@@ -6,6 +6,12 @@ def self.load_defaults
     # Check the table exists
     return unless super
 
+    Setting.transaction do
+      [
+        self.set('discovery_fact', _("The default fact name to use for the MAC of the system"), "macaddress"),
```
Comment (lzap) [inline app/models/setting/discovered.rb:11]: So we are changing from macaddress_eth0 to macaddress? Just to confirm...

**Comment by lzap** [comment]
WFM

File: app/models/setting/discovered.rb (hunk 1 above)
Comment (GregSutcliffe) [inline app/models/setting/discovered.rb:11]: We're changing the default to "macaddress", to fit with the new RPM images. I'll merge this and do a new gem release once we feel the RPM's are ready for prime-time :)

Comment (lzap) [inline app/models/setting/discovered.rb:11]: Ok, except the extra 500 error when rebooting (fix is going soon) I think we are fine. We need more testing from the community tho.

Comment (GregSutcliffe) [inline app/models/setting/discovered.rb:11]: Great, we can move the RPM image to the primary method of using Discovery for 1.4.

Comment (lzap) [inline app/models/setting/discovered.rb:11]: Right, I was thinking about versioning scheme. Do we want to follow
Foreman core, or keep very own version (different to TCL)? Looks like a
mess now.

LZ

On Tue, Jan 07, 2014 at 03:19:17AM -0800, Greg Sutcliffe wrote:

> > @@ -6,6 +6,12 @@ def self.load_defaults
> >      # Check the table exists
> >      return unless super
> > -    Setting.transaction do
> > -      [
> > -        self.set('discovery_fact', _("The default fact name to use for the MAC of the system"), "macaddress"),
> 
> Great, we can move the RPM image to the primary method of using Discovery for 1.4.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman_discovery/pull/46/files#r8690706

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

Comment (GregSutcliffe) [inline app/models/setting/discovered.rb:11]: I tend to release a new version of the plugin with each core release, as usually things have changed. I think we can use our own version numbers in the plugin though.

**Comment by GregSutcliffe** [comment]
Merged, thanks for the reviews guys

**Pull request closed without merging**